src/back/wgsl/writer.rs

```diff
@@ -839,6 +1127,17 @@ fn scalar_kind_str(kind: ScalarKind) -> &'static str {
     }
 }
 
+fn scalar_value_str(value: ScalarValue) -> String {
```

Comment:
let's have this as a method on `self` instead, so that we can write directly to `self.out` instead of allocating this temporary string